src/main/scala/com/wavesplatform/UtxPool.scala

```diff
@@ -77,6 +79,28 @@ class UtxPool(time: Time,
       })
   }
 
+  private def checkNotBlacklisted(tx: Transaction): Either[ValidationError, Unit] = {
+    val sender: Option[String] = tx match {
+      case x: SignedTransaction => Some(x.sender.address)
+      case x: PaymentTransaction => Some(x.sender.address)
+      case x: GenesisTransaction => x.creator.map(_.address)
+      case _ => None
+    }
+
+    val recipient: Option[String] = tx match {
```

Comment:
please keep it w/o .stringRepr because it's computation.
I also suggest skip completely all checks if `utxSettings.blacklistSrcAddresses` is empty